sockeye/arguments.py

```diff
@@ -148,11 +148,41 @@ def add_model_parameters(params):
                               type=int_greater_or_equal(0),
                               default=50000,
                               help='Maximum vocabulary size. Default: %(default)s.')
+    model_params.add_argument('--num-words-source',
+                              type=int_greater_or_equal(0),
+                              default=None,
+                              help='Maximum source vocabulary size. Overrides --num-words. Default: %(default)s')
+    model_params.add_argument('--num-words-target',
+                              type=int_greater_or_equal(0),
+                              default=None,
+                              help='Maximum target vocabulary size. Overrides --num-words. Default: %(default)s')
     model_params.add_argument('--word-min-count',
                               type=int_greater_or_equal(1),
                               default=1,
                               help='Minimum frequency of words to be included in vocabularies. Default: %(default)s.')
 
+    model_params.add_argument('--char-seq-encoder',
```

Comment:
We could call it ConvolutionalEmbeddingEncoder to avoid confusion with Facebook's convolutional encoder.  We could pack everything into the --encoder arg if we have a good idea for what we want that to look like.  Two obvious options are:

1. Let users list any encoders they like in sequence, basically moving get_encoder's functionality to the command line:
`--encoder embed birnn`
`--encoder embed conv-embed birnn`
`--encoder embed conv-embed transformer`
Here we have to keep track of when to add batch2time and other "helper" encoders.

2. Have some presets for users:
`--encoder rnn`
`--encoder rnn-with-conv-embed`
`--encoder transformer`
`--encoder transformer-with-conv-embed`
Here we manage everything.

My suggestion would be to go with 2 for now and move toward 1.  Thoughts?